Add etl/ directory with documentation for marshmallow

hancush opened the pull request:
## Overview

See title.

Handles #63

## Testing Instructions

* View the rendered markdown and ensure that it looks good.
* Click the links and confirm they resolve.

jeancochrane commented on etl/README.md:
```diff
@@ -0,0 +1,21 @@
+# ETL
+
+<blockquote>
+<p>ETL refers to the general process of:</p>
+
+<p>
+    <ol>
+        <li>taking raw <strong>source data</strong> (<em>"Extract"</em>)</li>
+        <li>doing some stuff to get the data in shape, possibly involving intermediate <strong>derived files</strong> (<em>"Transform"</em>)</li>
+        <li>producing <strong>final output</strong> in a more usable form (for <em>"Loading"</em> into something that consumes the data - be it an app, a system, a visualization, etc.)</li>
+    </ol>
+</p>
+</blockquote>
```
This works, but is there a reason you chose to use HTML tags instead of the Markdown quote operator `>`?

jeancochrane replied:
Also, is there a reason for this to be an un-attributed quote instead of normal body text?

hancush replied:
AFAIK, formatting for multi-paragraph block quotes gets messed up, and I wanted to preserve it.

jeancochrane commented on etl/examples/dedupe_api_schemas.py:
```diff
@@ -0,0 +1,318 @@
+
+    @validates('cluster_id')
+    def validate_cluster_id(self, value, **kwargs):
+        entity_map = 'change_log_{}'.format(self.project.id)
+
+        entity_query = '''
+            SELECT EXISTS(
+              SELECT 1 FROM "{entity_map}"
+              WHERE entity_id = :entity_id
+            )
+        '''.format(entity_map=entity_map)
```
On second thought, I'm nervous about exposing our schema like this. I'm also skeptical that this file will be useful to devs who are not familiar with Dedupe; even with detailed knowledge of the data model and API spec, I still found it a challenging read. How about we remove this file (and squash on merging), add a link to the source code to the documentation, and add a note saying that devs who don't have access to the source can request a gist from a Dedupe.io dev? I'm expecting that the time this file will come in handy the most for me is if I need to task a dev with pulling off a tricky Marshmallow move and want to give them an example to work from, and in that case creating a gist for them based on the source code wouldn't be substantially more work than directing them to this file.

hancush commented:
@jeancochrane Unquoted the README, removed the example and added guidance to request an example from a member of the `dedupeio` organization on GitHub. Ready for re-review!
